packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,393 @@
+        return this.exit(err);
+      });
+  }
+
+  async inferRepositoryType() {
+    let result = await this._isConnectorOfType(
+      KEY_VALUE_CONNECTOR,
+      this.artifactInfo.dataSourceClassName,
+    );
+
+    if (result) {
+      this.artifactInfo.repositoryTypeClass = KEY_VALUE_REPOSITORY;
+      this.artifactInfo.defaultTemplate = REPOSITORY_KV_TEMPLATE;
+    } else {
+      this.artifactInfo.repositoryTypeClass = DEFAULT_CRUD_REPOSITORY;
+      this.artifactInfo.defaultTemplate = REPOSITORY_CRUD_TEMPLATE;
+    }
+
+    // assign the data source name to the information artifact
+    let dataSourceName = this.artifactInfo.dataSourceClassName
+      .replace('Datasource', '')
+      .toLowerCase();
+
+    Object.assign(this.artifactInfo, {
+      dataSourceName: dataSourceName,
+    });
+  }
+
+  async promptModels() {
+    let modelList;
+    try {
+      modelList = await utils.getArtifactList(
+        this.artifactInfo.modelDir,
+        'model',
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    if (_.isEmpty(modelList)) {
+      return this.exit(
+        `${ERROR_NO_MODELS_FOUND} ${this.artifactInfo.modelDir}.
+        ${chalk.yellow(
+          'Please visit http://loopback.io/doc/en/lb4/Repository-generator.html for information on how models are discovered',
+        )}`,
+      );
+    }
+
+    return this.prompt([
+      {
+        type: 'checkbox',
+        name: 'modelNameList',
+        message: PROMPT_MESSAGE_MODEL,
+        choices: modelList,
+        when: this.artifactInfo.modelNameList === undefined,
+        default: modelList[0],
+      },
+    ])
+      .then(props => {
+        debug(`props: ${inspect(props)}`);
+        Object.assign(this.artifactInfo, props);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during prompt for repository variables: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptModelId() {
+    if (_.isEmpty(this.artifactInfo.modelNameList)) {
+      return this.exit(new Error(`${ERROR_NO_MODEL_SELECTED}`));
+    } else {
+      // iterate thru each selected model, infer or ask for the ID type
+      for (let item of this.artifactInfo.modelNameList) {
+        debug(`iterating thru ${item}`);
+        this.artifactInfo.modelName = item;
+        let modelPropertyList = await this._getModelFields(item);
+
+        const prompts = [
+          {
+            type: 'list',
+            name: 'propertyName',
+            message: `Please, select the ID property for ${item} from the list `,
+            choices: modelPropertyList,
+            default: modelPropertyList[0],
+          },
+        ];
+
+        let idType = await this._getModelIdType(item);
+
+        if (idType === 'none') {
+          const answer = await this.prompt(prompts);
+          this.artifactInfo.idType = `typeof ${item}.prototype.${
+            answer.propertyName
+          }`;
+        } else {
+          this.artifactInfo.idType = idType;
```

Comment:
Please use `typeof ${item}.prototype${idType}` in this branch too.

The benefit is that when people rename models and/or change the `id` property type in their application later in the future, the Controller code will automatically track those changes.

When you put a hard-coded ID type in the generated controller source, users have to update that ID type when it changes, which can be tedious.